src/auction/src/orders.mo

```diff
@@ -346,12 +351,33 @@ module {
       switch (cancellations) {
         case (null) {};
         case (? #all(null)) {
+          switch (expectedSessionNumber) {
+            case (?sn) {
+              for ((asset, aid) in Vec.items(assets.assets)) {
+                if (asset.sessionsCounter != sn) {
+                  return #err(#cancellation({ error = #SessionNumberMismatch(aid); index = 0 }));
```

Comment:
Why can we set the index to 0 here?

Reply:
No more index in SessionNumberMismatch error